src/lib/tree/data-source/nested-data-source.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {CollectionViewer, DataSource} from '@angular/cdk/collections';
+import {Observable} from 'rxjs/Observable';
+import {merge} from 'rxjs/observable/merge';
+import {map} from 'rxjs/operators/map';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+
+/**
+ * Data source for nested tree.
+ *
+ * The data source for nested tree doesn't have to consider node flattener, or the way to expand
+ * or collapse. The expansion/collapsion will be handled by TreeControl and each non-leaf node.
+ */
+export class MatTreeNestedDataSource<T> implements DataSource<T> {
+  _renderedData = new BehaviorSubject<T[]>([]);
+  get renderedData(): T[] { return this._renderedData.value; }
```

Comment:
I don't quite follow what `renderedData` is needed/used for

Reply:
Removed. `data` is enough